### PR TITLE
Refactor sql tables

### PR DESCRIPTION
**Pull request by TCprof**

### CODE REVIEW

**Review by stephenmcguigan-kainos — DISMISSED** [review]

File: SQLScripts/CreateTables_ATeam.sql [hunk 1]
```diff
@@ -30,33 +30,32 @@ CREATE TABLE `band` (
...
-CREATE TABLE competency_element(
-    competencyElementID TINYINT AUTO_INCREMENT,
-    competencyID TINYINT not null,
-    elementName varchar (40),
-    bandID TINYINT,
-    elementDescription text,
-    PRIMARY KEY (competencyElementID),
-    CONSTRAINT competency_element_to_competency_FK FOREIGN KEY (competencyID) REFERENCES competency (competencyID),
-    CONSTRAINT band_to_competency_element_FK FOREIGN KEY (bandID) REFERENCES band (bandID)
+CREATE TABLE 'competency_element'(
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/CreateTables_ATeam.sql:39]: wrong marks
```suggestion
CREATE TABLE `competency_element`(
```

File: SQLScripts/CreateTables_ATeam.sql [hunk 2]
```diff
@@ -30,33 +30,32 @@ CREATE TABLE `band` (
...
-  `responsibilityId` smallint NOT NULL,
-   PRIMARY KEY (`roleId`,`responsibilityId`),
-   CONSTRAINT `role_responsibilities_ibfk_1` FOREIGN KEY (`responsibilityId`) REFERENCES `responsibilities` (`responsibilityId`)
- );
-
-CREATE TABLE competency(
-    competencyID TINYINT PRIMARY KEY AUTO_INCREMENT,
-    competencyName varchar (40),
-    competencyDescription text
+CREATE TABLE 'competency'(
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/CreateTables_ATeam.sql:33]: wrong marks
```suggestion
CREATE TABLE `competency`(
```

**Review by stephenmcguigan-kainos — CHANGES_REQUESTED** [review]
Could we put the band and capability tables in front of the role table so the script could be run in one big go

File: SQLScripts/InsertData_ATeam.sql [hunk 3]
```diff
@@ -4,28 +4,34 @@ use Kainos_A;
 INSERT INTO band(bandName, bandExpectations)
 VALUES ("Apprentice", ""), ("Trainee", ""), ("Associate", ""),("Senior Associate", ""), ("Consultant", ""), ("Manager",""), ("Principle", ""), ("Leadership Community", "");
 
-insert into capability (capability) VALUES(("Engineering"),
-											("Platforms"));
+
+
+INSERT INTO capability (capability) VALUES(("Engineering"),("Platforms"));
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/InsertData_ATeam.sql:9]: this thinks only trying to put in one ow cause of the outside ()s
```suggestion
INSERT INTO capability (capability) VALUES("Engineering"),("Platforms");
```

File: SQLScripts/InsertData_ATeam.sql [hunk 4]
```diff
@@ -4,28 +4,34 @@ use Kainos_A;
...
+
+
 
 INSERT INTO roles (`roleName`, `roleDesc`, `datePosted`, `hours`, `location`, `bandID`, `capabilityID`, `link`)
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/InsertData_ATeam.sql:13]: role not roles
```suggestion
INSERT INTO role (`roleName`, `roleDesc`, `datePosted`, `hours`, `location`, `bandID`, `capabilityID`, `link`)
```

File: SQLScripts/CreateTables_ATeam.sql [hunk 5]
```diff
@@ -30,33 +30,32 @@ CREATE TABLE `band` (
...
-   CONSTRAINT `role_responsibilities_ibfk_1` FOREIGN KEY (`responsibilityId`) REFERENCES `responsibilities` (`responsibilityId`)
- );
-
-CREATE TABLE competency(
-    competencyID TINYINT PRIMARY KEY AUTO_INCREMENT,
-    competencyName varchar (40),
-    competencyDescription text
+CREATE TABLE `competency`(
+    `competencyID` tinyint PRIMARY KEY AUTO_INCREMENT,
+    `competencyName` varchar (40),
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/CreateTables_ATeam.sql:35]: data you insert is being truncated maybe increase the limit

File: SQLScripts/CreateTables_ATeam.sql [hunk 6]
```diff
@@ -30,33 +30,32 @@ CREATE TABLE `band` (
...
-    elementName varchar (40),
-    bandID TINYINT,
-    elementDescription text,
-    PRIMARY KEY (competencyElementID),
-    CONSTRAINT competency_element_to_competency_FK FOREIGN KEY (competencyID) REFERENCES competency (competencyID),
-    CONSTRAINT band_to_competency_element_FK FOREIGN KEY (bandID) REFERENCES band (bandID)
+CREATE TABLE `competency_element`(
+    `competencyElementID` tinyint AUTO_INCREMENT,
+    `competencyID` tinyint NOT NULL,
+    `elementName` varchar (40),
```
Comment (stephenmcguigan-kainos) [inline SQLScripts/CreateTables_ATeam.sql:42]: data you insert is being truncated maybe increase the limit

**Review by stephenmcguigan-kainos — CHANGES_REQUESTED** [review]

File: SQLScripts/CreateTables_ATeam.sql [hunk 7]
```diff
@@ -2,13 +2,26 @@ CREATE DATABASE Kainos_A;
...
-CREATE TABLE `roles` (
+CREATE TABLE `capability` (
+  `capabilityID` tinyint NOT NULL AUTO_INCREMENT,
+  `capability` varchar(40) DEFAULT NULL,
+  PRIMARY KEY (`capabilityID`)
+);
+
+CREATE TABLE `band` (
+  `bandID` tinyint NOT NULL AUTO_INCREMENT,
+  `bandName` varchar(40) NOT NULL,
```
Comment (KristinaGeddis) [inline SQLScripts/CreateTables_ATeam.sql:13]:
```suggestion
  `bandName` varchar(40) NOT NULL
```

File: SQLScripts/CreateTables_ATeam.sql [hunk 8]
```diff
@@ -2,13 +2,26 @@ CREATE DATABASE Kainos_A;
...
+  `bandExpectations` text,
```
Comment (KristinaGeddis) [inline SQLScripts/CreateTables_ATeam.sql:14]:
```suggestion
```
We don't have this anymore

**Review by KristinaGeddis — DISMISSED** [review]

File: SQLScripts/CreateTables_ATeam.sql (hunk 7 above)
Comment (TCprof) [reply SQLScripts/CreateTables_ATeam.sql]: need a comma on the field before a primary key declaration

**Review by stephenmcguigan-kainos — APPROVED** [review]

**Review by AmamMcMam — APPROVED** [review]

**Pull request merged**